### ISSUE TITLE
ipv6error ipv6neigh ipv6router

### ISSUE THREAD

**Issue by ktsaou** [ktsaou/firehol#65]
@philwhineray I have the impression that ipv6error is not usefull any more.

in `close_interface()` and `close_router()` we now have:

```bash
	rule chain "in_${work_name}" state RELATED proto icmpv6 action ACCEPT || return 1
	rule chain "out_${work_name}" state RELATED proto icmpv6 action ACCEPT || return 1
```

In my IPv4 tests I have seen that ICMP errors are reported as RELATED, FireHOL does not load the icmp conntrack helper, but it seems that the netfilter does thiis by itself.

So, can you tell if ipv6error is still required to be defined by the user?

Also, I wonder if we could also handle ipv6neigh and ipv6router automatically/ Why not adding ipv6neight and ipv6router for all ipv6 interfaces?

**Comment by philwhineray** [comment]
@ktsaou Short answer I am not sure.

I will look into the ipv6error() situation. It probably is not now doing anything useful based on close_interface() and close_router(). It was implemented to meet the [recommendations of RFC 4890 ](http://firehol.org/guides/icmpv6-recommendations) so I guess I want to double-check that every RELATED icmpv6 packet should be permitted even if it is not an error (maybe not), and that every icmpv6 error type will be marked as RELATED (seems very likely).

I did consider adding ipv6neigh and  ipv6router to all interfaces. I think it would be safe to allow neighbour discovery (as well as necessary for correct operation).

I am not so sure about router discovery though, since this presents a possible security risk: unlike IPv4 where your default route will be set once via DHCP, the default stateless configuration with RA/RD runs all the time, so anyone can come on your network and try to hijack your route unless you disable them or at least use an optional rule parameter to filter them.

Since my use-case includes bridges, I still needed an ipv6neigh to allow these in router statements, so in the end I was a coward and left everything to the end user.

Probably v3 is a good time to make use of `firehol-defaults.conf` and insert these rules by default but to let the user switch them off and take control manually.
